common/hal/hal_gpio.c

```diff
@@ -135,6 +136,47 @@ int gpio_set(uint8_t gpio_num, uint8_t status)
 	}
 }
 
+uint8_t get_gpio_direction(uint8_t gpio_idx)
+{
+	uint32_t gpio_dir = sys_read32(GPIO_GROUP_REG_ACCESS[gpio_idx / 32] + 0x4);
+
+	return GETBIT(gpio_dir, gpio_idx % 32);
+}
+
+uint8_t get_gpio_interrupt_enable(uint8_t gpio_idx)
+{
+	uint32_t gpio_intr_en = sys_read32(GPIO_GROUP_REG_ACCESS[gpio_idx / 32] + 0x8);
+
+	return GETBIT(gpio_intr_en, gpio_idx % 32);
+}
+
+uint8_t get_gpio_interrupt_type(uint8_t gpio_idx)
+{
+	uint32_t gpio_intr_type = sys_read32(GPIO_GROUP_REG_ACCESS[gpio_idx / 32] + 0x10);
+
+	return GETBIT(gpio_intr_type, gpio_idx % 32);
+}
+
+uint8_t get_gpio_interrupt_trigger_mode(uint8_t gpio_idx)
+{
+	uint8_t trigger_mode = 0;
+	uint32_t gpio_intr_dual_edge = sys_read32(GPIO_GROUP_REG_ACCESS[gpio_idx / 32] + 0x14);
+	uint32_t gpio_intr_rsing_falling_edge =
+		sys_read32(GPIO_GROUP_REG_ACCESS[gpio_idx / 32] + 0xC);
+
+	if (gpio_intr_dual_edge & BIT(gpio_idx % 32)) {
+		trigger_mode = 0x2; //both edge
+	} else {
+		if (gpio_intr_rsing_falling_edge & BIT(gpio_idx % 32)) {
+			trigger_mode = 0x1; //rising edge
+		} else {
+			trigger_mode = 0x0; //falling edge
+		}
+	}
+
+	return trigger_mode;
```

Comment:
Can we make this a little more generic in case we need to support different boards in the future?

Preferably we wouldn't used hard coded values here and also we would use some of the Zephyr GPIO APIs that might help here.

Reply:
https://docs.zephyrproject.org/2.6.0/reference/peripherals/gpio.html
There is no Zephyr GPIO API for getting gpio direction and gpio interrupt information. After discussing with Aspeed, they suggested for maintaining a GPIO status table and get the information from the table. Currently ,we are studying how to implement this feature. Do you have any concern?

Reply:
No concern, thank you for looking into this.

Reply:
Modify this commit to a function of align ti BIC GPIO number. And the feature of GPIO will be added in the later commit.